TOOLS_OVERVIEW.md

```diff
@@ -0,0 +1,228 @@
+
+Details of all the options can be seen by running
+```
+goto-analyzer --help
+```
+
+## goto-cc
+
+This is a compiler that also adds goto program information to the compiled
+programs. Note that `goto-cc` is the compiler and linker while `goto-gcc` and
+`goto-ld` are symbolic links to `goto-cc`. The additional object code is
+used as the internal representation for `cbmc` and related tools. These can
+also be extracted and used themselves.
+
+Further information on `goto-cc` can be found:
+- [Developer Tutorial](http://cprover.diffblue.com/tutorial.html) with
+some very simple examples and notes.
+- [goto-cc](http://cprover.diffblue.com/group__goto-cc.html) information
+on the `goto-cc` compilers
+- [goto Programs](http://cprover.diffblue.com/group__goto-programs.html)
+for information on goto programs and how they are used.
+
+
+## goto-diff
+
+Provides a variety of difference checks between two goto programs (produced
+by `goto-cc`). This invokes some of the cbmc tools to convert the goto
+program and then determine which functions are added/removed/changed.
+
+Details of all the options can be seen by running
+```
+goto-diff --help
+```
+this includes both options for the difference, and options for the goto
+program instrumentation.
+
+
+## goto-harness
+
+This is a tool for creating a harness around a (part of a) goto program that
+can then be analysed inside the harness. Harnesses can be either function
+call environments, or memory snapshots. 
+
+Documentation on `goto-harness` can be found
+[here](http://cprover.diffblue.com/md__home_travis_build_diffblue_cbmc_doc_architectural_goto-harness.html)
+including details and examples.
+
+Details of all the options can be seen by running
+```
+goto-harness --help
+```
+
+
+## goto-instrument
+
+This is a collection of tools for analysing and modifying goto programs 
+(programs created with #goto-cc). Generally these take a goto program 
+and output another goto program.
+
+Further examples and documentation can be found:
+- [goto-instrument](http://cprover.diffblue.com/group__goto-instrument.html)
+has an overview of `goto-instrument` and a small tutorial example.
+- [Developer Tutorial](http://cprover.diffblue.com/tutorial.html) has high
+level overview and some example commands for `goto-instrument`.
+
+
+## janalyzer
+
+This provides a way to access and invoke various forms of analysis on
+Java programs.
+
+Documentation useful for this tool can be found:
+- [Analysis Information](http://cprover.diffblue.com/group__analyses.html)
+
+Details of all the options can be seen by running
+```
+janalyzer --help
+```
+
+## java-unit
+
+Runs Java unit tests. For more details use
+```
+java-unit --help
+```
+Default behaviour is to only show failed test cases.
+
+
+## jbmc
+
+This is the main analysis engine the performs the analysis
+of Java files using bounded model checking. This is a
+Java version of the `cbmc` tool (more documentation for
```

Comment:
Yes, it's a BMC tool for Java bytecode (similar to cbmc being a BMC tool for C).
It has been separated from CBMC in order to support separate compilation of C and Java frontends and tools. Also, JBMC's command line follows conventions of `java` so that programs can be analysed in the same way you would run them using the `java` executable. There also lots of additional options to support nondeterministic initialisation of variable size arrays, data structures and strings, control how to assert on exceptions etc.

Reply:
Updated text for this (and added some clear phrases from Peter).